### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by techEmprez**
### Project requirements
-  Create a method called `current_user` in your ApplicationController to make current user data available to all controllers.
    -  It will return the first user from the database.
    -  Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
    - Creates a Post on behalf of the `current_user` (use the method that you created in your ApplicationController).
    - Create a comment on behalf of the `current_user` (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/views/comments/new.html.erb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 <div>
-   
+    <h2> Create Comment </h2>
+    <%= form_with model: @comment, url: user_post_comments_path, data: { turbo: false} do |form| %>
+        
         <div class="mb-3">
             <%= form.label :text, class: "form-label" %>
             <%= form.text_field :text, class: "form-control", id: "exampleFormControlInput1", placeholder: "Post Comment" %>
```
Comment (zieeco) [inline app/views/comments/new.html.erb]: - Hey, the User can't add a comment as there appears to be no form in your UI to integrate that. Also, one can't create a post as there's no form to add this. please, kindly ensure to fix this by adding a form to create a `Post`, add a `Comment`, and add a `Like`. These features are missing in your app. You should have a form like this:

image

image
image

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @techEmprez,





Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️
- Nice trying to implement the form. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/comments/new.html.erb (hunk 1 above)
Comment (techEmprez) [reply app/views/comments/new.html.erb]: All done

File: app/views/comments/new.html.erb [hunk 2]
```diff
@@ -1,11 +1,8 @@
...
-    <%= form_with model: @comment, url: user_post_comments_path, data: { turbo: false} do |form| %>
-        
-        <div class="mb-3">
-            <%= form.label :text, class: "form-label" %>
-            <%= form.text_field :text, class: "form-control", id: "exampleFormControlInput1", placeholder: "Post Comment" %>
-        </div>
-        <%= form.submit "Post Comments", class:"btn btn-primary" %>
-    <% end %>
-</div>
+<%= form_with model: @comment, url: user_post_comments_path do |form| %>
```
Comment (rustam-tolipov) [inline app/views/comments/new.html.erb:1]: **You did great job!**

- [ ] Your app displays another error after previous reviewers request. If I click `See more` button in users post page. It shows me this error. Kindly fix this error and resubit again :)

Screenshot 2023-05-14 at 00 11 14

Screenshot 2023-05-14 at 00 14 17

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi @techEmprez, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
1. Greate readme 💯 
2. Well-implemented code 💯 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/comments/new.html.erb (hunk 2 above)
Comment (techEmprez) [reply app/views/comments/new.html.erb:1]: all done

File: app/views/comments/new.html.erb [hunk 3]
```diff
@@ -1,11 +1,10 @@
-<div>
-    <h2> Create Comment </h2>
-    <%= form_with model: @comment, url: user_post_comments_path, data: { turbo: false} do |form| %>
-        
-        <div class="mb-3">
-            <%= form.label :text, class: "form-label" %>
-            <%= form.text_field :text, class: "form-control", id: "exampleFormControlInput1", placeholder: "Post Comment" %>
-        </div>
-        <%= form.submit "Post Comments", class:"btn btn-primary" %>
-    <% end %>
```
Comment (Munsa1) [inline app/views/comments/new.html.erb:10]: - [ ] As requested by the previous reviewer, kindly fix this error that comes about when the user clicks the `see more` button
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @techEmprez, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Passing Linters :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/comments/new.html.erb (hunk 3 above)
Comment (techEmprez) [reply app/views/comments/new.html.erb:10]: ALL DONE

Comment (techEmprez) [reply app/views/comments/new.html.erb:10]: Please reviewer, kindly run rails db:reset before reviewing
Thank you....

Please reach out via slack if need be as this is my last review... PLease

Thank you

**Review by MaganaAsiati — APPROVED** [review]
# Status: Approved [✔️](https://emojipedia.org/check-mark/)
## Hi @techEmprez , I'm @MaganaAsiati , your reviewer for this PR  :wave: 
Great work on making the changes requested by a previous reviewer 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 





## Highlights
- Descriptive pull request :heavy_check_mark:👨🏻‍💻
- Descriptive README file  :heavy_check_mark:👨🏻‍💻
- Great work using the right GitHub Flow.:heavy_check_mark:
- No Linter errors :heavy_check_mark:


## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] [OPTIONAL] Please make sure to make your commit messages more descriptive in the future.
- [ ] Check the comments under the review.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
Cheers and Happy coding! 🤓 👏👏👏

**Pull request merged**